torchvision/prototype/transforms/_meta.py

```diff
@@ -23,6 +23,10 @@ def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
 
 
 class ConvertImageDtype(Transform):
+    # Question: Why do we have both this and a ToDtype Transform? Is this due to BC? Ideally we could move people off
+    # from methods that did an implicit normalization of the values (like this one, or to_tensor). cc @vfdev-5
+    # If that's the case, we should move to _legacy and add deprecation warnings from day one to push people to use
+    # the new methods.
```

Comment:
One idea that @datumbox and I discussed a while back is abolish the close correspondence of the dtype and the value range for images. Right now we assume the range `[0, 1]` for floating point images and `[0, torch.iinfo(dtype).max]` for integral images. A possible solution for this would be have a `value_range` meta data on the `features.Image` class. It can be set with the defaults from above so for a regular use case the user won't feel any difference. 

For example, let's say we have a uint8 image

```py
import torch
from torchvision.prototype import features

image = features.Image(torch.randint(0, 256, (3, 128, 128)))
```

By default `print(image.value_range)` would print `(0, 255)`. If we now convert the dtype like `image = image.to(torch.float32)`, printing the value range again would give `(0.0, 255.0)` rather than `(0.0, 1.0)` what happens through `ConvertImageDtype`.

This means, all transformations that implicitly rely on the value range like `Normalize` need to be adapted to use the explicit value instead, but we could get rid of `ConvertImageDtype` all together.

Reply:
I absolutely love this and 100% agree we should do it. I've previously asked @vfdev-5 to create an issue and see which methods of the low-level kernels should be expanded to receive a max_value. We can write this in a BC manner on the low-level kernels; if the `max_value` is not passed we do the previous behaviour. `Tensors` will continue making this assumption, but `Images` shouldn't.

I feel this is a critical part of the proposal that needs to be investigated prior implementing all other transforms. This is probably one of the few remained uninvestigated bits of the proposal.

Reply:
FYI #5502